src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -162,12 +179,11 @@ export class KernelProcess implements IKernelProcess {
 
             if (cancelWaiting.token.isCancellationRequested) {
                 traceError(stderrProc || stderr);
-                throw new Error(
-                    localize.DataScience.kernelDied().format(
-                        Commands.ViewJupyterOutput,
-                        (stderrProc || stderr).substring(0, 100)
-                    )
-                );
+                // If we have the python error message, display that.
+                const errorMessage =
+                    getErrorMessageFromPythonTraceback(stderrProc || stderr) ||
+                    (stderrProc || stderr).substring(0, 100);
+                throw new Error(localize.DataScience.kernelDied().format(Commands.ViewJupyterOutput, errorMessage));
```

Comment:
Better display of error messages, since we have logged the Python error in console, we can display the meaningful Python error message from the traceback.
(previously this used to be empty as error message is sent as part of other messages in the rx subject).